### PR TITLE
ISSUE-405 fixed query in ListAvailableRolesByResourceType method in r…

### PR DESCRIPTION
**Pull request by sbryzak**
So far this PR just fixes up the query, still working on the db_clean issue which _may_ be related to other test failures... the investigation continues.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-413-1`

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-413-2`

**Comment by sbryzak** [comment]
Since @alexeykazakov fixed the major problem with the tests, this PR is limited to fixing a minor bug in the role management service, while handling roles without any scopes.  

Test included.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-413-3`

File: authorization/role/model/role_management_model_service.go [hunk 1]
```diff
@@ -261,19 +261,20 @@ func (r *GormRoleManagementModelService) ListAvailableRolesByResourceType(ctx co
 	defer goa.MeasureSince([]string{"goa", "db", "role", "listAvailableRoles"}, time.Now())
 	var roleScopes []role.RoleScope
 
-	r.db = r.db.Debug()
+	//r.db = r.db.Debug()
```
Comment (sbose78) [inline authorization/role/model/role_management_model_service.go]: let's remove the comment?

Comment (sbryzak) [reply authorization/role/model/role_management_model_service.go]: Oops, done.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-413-4`

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-413-5`

**Pull request merged**